src/SFA.DAS.ApprenticePortal/Application/Homepage/Queries/GetApprenticeHomepageQueryHandler.cs

```diff
@@ -0,0 +1,41 @@
+﻿using MediatR;
+using SFA.DAS.ApprenticePortal.Application.Extensions;
+using SFA.DAS.ApprenticePortal.InnerApi.ApprenticeAccounts.Requests;
+using SFA.DAS.ApprenticePortal.InnerApi.ApprenticeCommitments.Requests;
+using SFA.DAS.ApprenticePortal.Models;
+using SFA.DAS.SharedOuterApi.Configuration;
+using SFA.DAS.SharedOuterApi.Interfaces;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.ApprenticePortal.Application.Homepage.Queries
+{
+    public class GetApprenticeHomepageQueryHandler : IRequestHandler<GetApprenticeHomepageQuery, GetApprenticeHomepageQueryResult>
+    {
+        private readonly IApprenticeAccountsApiClient<ApprenticeAccountsApiConfiguration> _accountsApiClient;
+        private readonly IApprenticeCommitmentsApiClient<ApprenticeCommitmentsApiConfiguration> _commitmentsApiClient;
+
+        public GetApprenticeHomepageQueryHandler(
+            IApprenticeAccountsApiClient<ApprenticeAccountsApiConfiguration> accountsApiClient,
+            IApprenticeCommitmentsApiClient<ApprenticeCommitmentsApiConfiguration> commitmentsApiClient)
+            => (_accountsApiClient, _commitmentsApiClient) = (accountsApiClient, commitmentsApiClient);
+
+        public async Task<GetApprenticeHomepageQueryResult> Handle(GetApprenticeHomepageQuery request, CancellationToken cancellationToken)
+        {
+            var (_apprentice, _apprenticeships) = await TaskEx.AwaitAll(
+                Task.FromResult(_accountsApiClient.Get<Apprentice>(new GetApprenticeRequest(request.ApprenticeId))).Result,
+                Task.FromResult(_commitmentsApiClient.Get<GetApprenticeApprenticeshipsResult>(new GetApprenticeApprenticeshipsRequest(request.ApprenticeId))).Result
+                );
```

Comment:
This doesn't quite look right to me, The layout I was expecting was as follows

var (_apprentice, _apprenticeships) = await TaskEx.AwaitAll(
                _accountsApiClient.Get<Apprentice>(new GetApprenticeRequest(request.ApprenticeId)),
                _commitmentsApiClient.Get<GetApprenticeApprenticeshipsResult>(new GetApprenticeApprenticeshipsRequest(request.ApprenticeId)).Result
                );
                
 This code looks like it's evaluating the task prior to passing the result to the TaskEx.AwaitAll function.

Reply:
Yes, I don't think those `Task.FromResult` calls are needed.  They were in the example just to turn an `int` into a `Task<int>`, but the `Get` functions already return `Task`s.

I don't think there's any need for `.Result` in this calling code.
```
            var (_apprentice, _apprenticeships) = await TaskEx.AwaitAll(
                _accountsApiClient.Get<Apprentice>(new GetApprenticeRequest(request.ApprenticeId)),
                _commitmentsApiClient.Get<GetApprenticeApprenticeshipsResult>(new GetApprenticeApprenticeshipsRequest(request.ApprenticeId))
                );
```

Reply:
Updated as discussed